docs/rules/0191/php-namespace.md

```diff
@@ -0,0 +1,107 @@
+---
+rule:
+  aip: 191
+  name: [core, '0191', php-namespace]
+  summary: The `option php_namespace` annotation should be idiomatic if set.
+permalink: /191/php-namespace
+redirect_from:
+  - /0191/php-namespace
+---
+
+# PHP namespace annotation
+
+This rule enforces that if a proto file for a public API surface sets
+`option php_namespace`, that it uses language idiomatic conventions, as
+mandated in [AIP-191][].
+
+## Details
+
+This rule looks at each proto file, and complains if the `php_namespace` file
+annotation uses anything other than upper camel case, or includes characters
+other than letters, numbers, and `.`.
+
+It also ensures that versions with stability (e.g. `V1beta1`) are capitalized
+appropriately.
+
+## Examples
+
+### Case
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+syntax = "proto3";
+
+package google.example.v1;
+
+option php_namespace = "google\\example\\v1";  // Should be UpperCamelCase.
+```
+
+```proto
+// Incorrect.
+syntax = "proto3";
+
+package google.example.v1;
+
+option php_namespace = "Google::Example::V1";  // Separator should be backslash.
+```
+
+**Correct** code for this rule:
+
+```proto
+// Correct.
+syntax = "proto3";
+
+package google.example.v1;
+
+option php_namespace = "Google\\Example\\V1";
+```
+
+### Versions with stability
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+syntax = "proto3";
+
+package google.example.v1beta1;
+
+option php_namespace = "Google\\Example\\V1Beta1"; // Should be V1beta1.
+```
+
+**Correct** code for this rule:
+
+```proto
+// Correct.
+syntax = "proto3";
+
+package google.example.v1beta1;
+
+option php_namespace = "Google\\Example\\V1beta1";
+```
+
+## Known issues
+
+This rule will improperly complain if it encounters an acronym. For example, it
+will complain about `Google\Cloud\AutoML\V1`, preferring `AutoMl`. This lint
+rule **should** be disabled in this case.
```

Comment:
I would personally disagree with this. I think special casing for abbreviations introduces a human component into deciding these values, and in an optimal world, users don't have to set these values at all and the tooling does it automatically for them. I think making a human decide what is and isn't an abbreviation introduces issues, and it would be better just to have the abbreviation seem a little out of place with the option value.

Reply:
This might be true. Certainly the rule **may** be disabled, though (we have existing APIs that use this case). Not going to change it here since there are two equivalent files, but in principle I am open to rounding down to may.

Reply:
I would round down to may at the least, although I'd go even further and say that abbreviations should not be an exception at all. Users want to trust the yes/no decision a linter makes and not have to make manual interventions, especially at organization-scale.

Reply:
Will follow-up in another PR.